src/coreclr/src/jit/lclvars.cpp

```diff
@@ -7339,6 +7339,14 @@ Compiler::fgWalkResult Compiler::lvaStressLclFldCB(GenTree** pTree, fgWalkData*
             return WALK_SKIP_SUBTREES;
         }
 
+        // The assert in the second pass below, requires that these types match, or we have a TYP_BLK
+        //
+        if ((varDsc->lvType != lcl->gtType) && varDsc->lvType != TYP_BLK)
```

Comment:
Nit: parens on the second part of the conditional...

Reply:
Updated with extra parens in both places